added support for yaml language files

juumixx opened the pull request:
Fall back to yaml when json file is not found for translation.
This allows to add comments (for example blocks) in the translations.

ilteoood commented:
I REALLY LIKE IT! Good job!
Just one thing: can you edit the example app to provide a demostration of this new functionality?

juumixx commented:
Thanks. I will add an example as soon as I got rid of the type errors.

juumixx commented:
I fixed the type issues. The yaml now gets a recursive conversion treatment, so it can be used with any levels deep objects.
I updated the example to contain both json and yaml files.

ilteoood commented on lib/flutter_i18n.dart:
```diff
@@ -37,13 +39,32 @@ class FlutterI18n {
-    var localeString = await rootBundle.loadString('$_basePath/$fileName.json');
-    decodedMap = json.decode(localeString);
+  dynamic _convertYaml(final YamlNode node) {
+    if (node is YamlMap) {
```
In the documentation of the library I saw this: "This library currently doesn't support dumping to YAML. You should use json.encode from dart:convert instead"

So, can this function be implemented as follow?

return json.decode(json.encode(node))
